app/main.py

```diff
@@ -1 +1,31 @@
-# write your code here
+import json
+from decimal import Decimal
+
+
+def calculate_profit(file_name):
+    with open(file_name) as f:
+        trades = json.load(f)
+
+    earned_money = 0
+    matecoin_account = 0
+
+    for entry in trades:
+        matecoin_price = Decimal(entry["matecoin_price"])
+
+        if entry["bought"]:
+            earned_money -= Decimal(entry["bought"]) * matecoin_price
+            matecoin_account += Decimal(entry["bought"])
+
+        elif entry["sold"]:
```

Comment:
It would be better to use `if` here instead of `elif`. Because theoretically there can be both 'sold' and 'bought'.